web/src/components/netflow-traffic.tsx

```diff
@@ -92,32 +99,31 @@ export const NetflowTraffic: React.FC = () => {
   //update data & arguments on range changes
   React.useEffect(() => {
     //ensure range is set and different than previous value
-    if (range === null || range === previousRange) {
+    if (range === undefined || range === previousRange) {
       return;
     }
 
     setTRModalOpen(false);
     if (typeof range === 'number') {
       setQueryArguments({ timeRange: range.toString() });
-      removeQueryArguments(['startTime', 'endTime']);
+      removeQueryArguments([Q.StartTime, Q.EndTime]);
     } else if (typeof range === 'object') {
       setQueryArguments({ startTime: range.from.toString(), endTime: range.to.toString() });
-      removeQueryArguments(['timeRange']);
+      removeQueryArguments([Q.TimeRange]);
     } else {
-      removeQueryArguments(['startTime', 'endTime', 'timeRange']);
+      removeQueryArguments([Q.StartTime, Q.EndTime, Q.TimeRange]);
     }
-    tick();
-  }, [previousRange, range, tick]);
+  }, [previousRange, range]);
```

Comment:
Seems we have a problem here since it updates QueryArguments after the query

In case of moving from `custom time range` to `last X minutes` I get a 503 (Service Unavailable) error since the query contains both arguments `api/loki/flows?startTime=1640991600&endTime=1641682740&timeRange=300`

Reply:
I would move this to a useCallback function :
```
  const manageArguments = React.useCallback(() => {
    if (typeof range === 'number') {
      setQueryArguments({ timeRange: range.toString() });
      removeQueryArguments([Q.StartTime, Q.EndTime]);
    } else if (typeof range === 'object') {
      setQueryArguments({ startTime: range.from.toString(), endTime: range.to.toString() });
      removeQueryArguments([Q.TimeRange]);
    } else {
      removeQueryArguments([Q.StartTime, Q.EndTime, Q.TimeRange]);
    }
  }, [range]);
```

and call it from both `useEffect` and `tick` calls :
```
   ...
    setTRModalOpen(false);
    manageArguments();
  }, [manageArguments, previousRange, range]);
```

```
   ...
    setError(undefined);
    manageArguments();
    getFlows(getFiltersParams(filters, range))
   ...
```

Reply:
ok, I think the problem is that `getFiltersParams` combines both state data and URL data ; IMO it should only be working with state data.